Doc/tools/extensions/grammar_snippet.py

```diff
@@ -7,21 +12,38 @@
 from sphinx.util.docutils import SphinxDirective
 from sphinx.util.nodes import make_id
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import Any
+
+    from docutils.nodes import Node
+    from sphinx.application import Sphinx
+    from sphinx.util.typing import ExtensionMetadata
 
-class SnippetStringNode(nodes.inline):
+
+class snippet_string_node(nodes.inline):  # noqa: N801 (snake_case is fine)
     """Node for a string literal in a grammar snippet."""
 
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        rawsource: str = '',
+        text: str = '',
+        *children: Node,
+        **attributes: Any,
+    ) -> None:
```

Comment:
To match the constructor arguments of the superclass (this also allows for a small simplification later on)